xilem_core/src/view.rs

```diff
@@ -271,3 +272,65 @@ where
         message_result
     }
 }
+
+impl<V: ?Sized> ViewMarker for Rc<V> {}
+/// An implementation of [`View`] which only runs rebuild if the states are different
+impl<State, Action, Context, Message, V> View<State, Action, Context, Message> for Rc<V>
+where
+    Context: ViewPathTracker,
+    V: View<State, Action, Context, Message> + ?Sized,
+{
+    type Element = V::Element;
+    type ViewState = RcState<V::ViewState>;
+
+    fn build(&self, ctx: &mut Context) -> (Self::Element, Self::ViewState) {
+        let (element, view_state) = self.deref().build(ctx);
+        (
+            element,
+            RcState {
+                view_state,
+                dirty: false,
+            },
+        )
+    }
+
+    fn rebuild(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut Context,
+        element: Mut<Self::Element>,
+    ) {
+        // If this is the same value, or no rebuild was forced, there's no need to rebuild
```

Comment:
As a minimal change, this should be `and`. 
I think the main interesting part of this comment is the "rebuild was forced" statement.

This could have a longer comment here, but I'm not sure how much value that adds. I'd probably like to see some minimal documentation added to at least `RcState::dirty`; having the same comment repeated twice isn't great.

Reply:
I can readd the comment if it makes sense, I've added docs to either `MessageResult::RequestRebuild` and to the dirty flag.